src/include/migraphx/check_context.hpp

```diff
@@ -36,7 +38,22 @@ struct check_context
 {
     struct op : auto_register_op<op>
     {
-        std::string name() const { return "check_context::" + get_type_name<T>(); }
+        static std::string name()
+        {
+            const auto& op_type_name                      = get_type_name<T>();
+            const auto& split_name                        = split_string(op_type_name, ':');
+            std::vector<std::string> name_without_version = {"check_context"};
+            // op_type_name would contain internal namespace name with version_x_y_z
+            // remove version and construct op_name such as check_context::migraphx::gpu::context
+            std::copy_if(
+                split_name.begin(),
+                split_name.end(),
+                std::back_inserter(name_without_version),
+                [&](const auto& i) { return not i.empty() and not contains(i, "version"); });
+            static std::string op_name = join_strings(name_without_version, "::");
```

Comment:
This will only compute `join_strings` once, but it will still create `name_without_version` everytime, which is why i needs to all be in a seperate static function.

Reply:
was wondering exactly how it works with static. Thanks for the explanation. Clear now.